build.gradle

```diff
@@ -82,8 +81,19 @@ task downloadData {
     downloadSampleData([new Download(id: '628e8e3521cf45e9a28a12fe10c02c4d', name: 'naperville_imagery', ext: 'tpk')], 'naperville')
 }
 
+task copyNatives(type: Copy) {
+    description = "Copies the arcgis native libraries into the project build directory for development."
+    group = "build"
+    configurations.natives.asFileTree.each {
+        from(zipTree(it))
+    }
+    into "build/natives"
+}
+
 run {
+    dependsOn copyNatives
     mainClassName = project.hasProperty("mainClass") ? mainClass : 'com.esri.samples.map.display_map.DisplayMapSample'
+    environment "ARCGISRUNTIMESDKJAVA_100_5_0", "build/natives"
```

Comment:
@mbcoder and I were wondering if the environment variable set up should come before applying the mainClassName: if there are any timing issues then the main class name might be found before the runtime environment variable is. If timing could be an issue within the run task, then the environment variable set up should come before class name.

Reply:
The order of these properties does not matter. All task configurations just set things up. Gradle determines the correct order to do everything.